polytracker/src/polytracker/main.cpp

```diff
@@ -24,16 +24,19 @@ using json = nlohmann::json;
 extern int errno;
 std::string polytracker_forest_name = "";
 std::string polytracker_db_name = "";
-int byte_start = -1;
-int byte_end = -1;
+uint64_t byte_start = 0;
+uint64_t byte_end = 0;
 bool polytracker_trace = false;
 bool polytracker_trace_func = false;
 /**
  * Whether or not to save the input files to the output database
  */
 bool polytracker_save_input_file = true;
 decay_val taint_node_ttl = 0;
+// If this is empty, taint everything.
 std::string target_file = "";
+bool track_all_taint = false;
```

Comment:
Instead of having a bool to toggle between either tracking a single file or tracking all taints, I think we should instead get rid of this and also get rid of `target_file`. Instead, create a `std::unordered_set<std::string> target_files` containing all of the whitelisted files, generated by splitting `POLYPATH` delimited by colons. If `POLYPATH` is empty or undefined, then `target_files` will be empty. If `target_files` is empty, then implicitly default to tracking all files. See #6354.

Reply:
Done! There is now a `target_sources` set and some logic to parse strings in the form of `source1;source2` etc. If nothing is specified, everything is tainted.